src/main/java/com/semmle/jira/addon/config/upgrades/UpgradeTask4CreateConfigKeyField.java

```diff
@@ -0,0 +1,46 @@
+package com.semmle.jira.addon.config.upgrades;
+
+import com.atlassian.plugin.spring.scanner.annotation.export.ExportAsService;
+import com.atlassian.sal.api.message.Message;
+import com.atlassian.sal.api.upgrade.PluginUpgradeTask;
+import com.semmle.jira.addon.util.Constants;
+import com.semmle.jira.addon.util.JiraUtils;
+import java.util.Collection;
+import java.util.Collections;
+import org.springframework.stereotype.Component;
+
+@ExportAsService(PluginUpgradeTask.class)
+@Component
+public class UpgradeTask4CreateConfigKeyField implements PluginUpgradeTask {
+
+  private static final int BUILD_NUMER = 4;
+
+  @Override
+  public Collection<Message> doUpgrade() throws Exception {
+
+    if (JiraUtils.getConfigKeyCustomField() == null) {
+      throw new Exception(); // TODO should be a more suitable exception
```

Comment:
This should create the field if it is missing right?

Reply:
It should, but null is a possible return value if `CustomFieldManager::createCustomField` throws a `GenericEntityException`.

Reply:
Wasn't sure what best to do about all that?

Reply:
I think we should create the custom field and managed item in the UpgradeTask. In the upgrade task we can try to find if a field with the right name and owner already exists. If it doesn't exist we should create it, mark it as managed and marked it as owned by the "plugin". Regardless of whether the field existed or not, the upgrade task should store its ID in the configuration, so the place where we read from and write to the field can just use the ID directly without resolving the field by name.